### ISSUE TITLE
dap4_test_test_raw not working on cmake builds?

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#887]
When I check out current master, and do a cmake build, I am now seeing this failure:

```
158/165 Test #158: dap4_test_test_raw ....................***Failed    5.43 sec
        Start 159: C_tests_simple_xy_wr

```

This doesn't seem to happen with autotool builds. This seems new but I can't really say.

**Comment by WardF** [comment]
Not seeing this on our end; can you share how you are configuring with cmake, and I'll try to recreate?

**Comment by DennisHeimbigner** [comment]
Ed-
Try to locate a file in your cmake build directory named
.../Testing/Temporary/LastTest.log
It may contain more info about why it failed.

**Comment by edhartnett** [comment]
I am seeing this in every cmake build now. Perhaps there's a server down or slow?

Here's an example. I checkout the latest master and do:
```
cd build
CPPFLAGS='-I/usr/local_cmake/include' CFLAGS='-fPIC -g -Wall' LDFLAGS='-L/usr/local_cmake/lib' LD_LIBRARY_PATH='/usr/local_cmake/lib' cmake -D"BUILD_SHARED_LIBS=ON" -DHDF5_C_LIBRARY=/usr/local_cmake/lib/libhdf5.so -DHDF5_HL_LIBRARY=/usr/local_cmake/lib/libhdf5_hl.so -DHDF5_INCLUDE_DIR=/usr/local_cmake/include -D"CMAKE_PREFIX_PATH=/usr/local_cmake/" ..

CPPFLAGS='-I/usr/local_cmake/include' CFLAGS='-fPIC -g -Wall' LDFLAGS='-L/usr/local_cmake/lib' LD_LIBRARY_PATH='/usr/local_cmake/lib' make all test
```
And I get:

```
The following tests FAILED:
	158 - dap4_test_test_raw (Failed)
Errors while running CTest
Makefile:151: recipe for target 'test' failed
make: *** [test] Error 8
```

**Comment by edhartnett** [comment]
I am going to try this again later and see if I get the same result. I may be having some network issues here that are preventing a connection from being made in time, and so something's timing out.

**Comment by DennisHeimbigner** [comment]
I don't think that the dap4 tests are doing any remoter server accesses at the moment
because we have not found a home for our dap4 test server.
The term "raw" means that it is reading the raw on the wire data from a file.
I just ran locally and am not seeing any failures.
Also, you might insert this in the test_raw.sh file to get more output into the LastTest.log
file:
    export SETX=1

**Comment by edhartnett** [comment]
OK, my file system is being hammered by loads of CI testing, and has been all day. I've been playing with the new valgrind testing and it's pretty freaking slow. So I will try again when all my CI tests have stopped and see if I still get this problem.

**Comment by edhartnett** [comment]
OK, total false alarm. Shared library problem. It couldn't find the hdf5 HL library.

**Comment by edhartnett** [comment]
No, it was not a shared library problem. It is something I don't understand. ncdump is segfaulting for the tests in test_raw.sh.


```
ed@mikado:~/unidata/netcdf-c/build/dap4_test$ CPPFLAGS='-I/usr/local_cmake/include' CFLAGS='-fPIC -g -Wall' LDFLAGS='-L/usr/local_cmake/lib' LD_LIBRARY_PATH='/usr/local_cmake/lib' ./test_raw.sh 
test_raw.sh:
testing: test_anon_dim.2.syn
./test_raw.sh: line 45:   726 Segmentation fault      (core dumped) ${VG} ${NCDUMP} "${URL}" > ./results_test_raw/${f}.dmp
*** Fail: [dap4]file:///home/ed/unidata/netcdf-c/build/dap4_test/daptestfiles/test_anon_dim.2.syn
1,13d0
< netcdf test_anon_dim.2 {
< dimensions:
< 	_Anonymous4 = 4 ;
< variables:
< 	int vu32(_Anonymous4) ;
< 
< // global attributes:
< 		string :_dap4.ce = "/vu32[0:3]" ;
< 		:_DAP4_Little_Endian = 1UB ;
< data:
< 
<  vu32 = -1169720286, -920699049, -2088732436, 1060190036 ;
< }
*** Fail: diff test_anon_dim.2.syn.dmp
```

How exactly do you guys build with cmake, specifying the directories for HDF5? That is, can you send me what you are using from the command line?

**Comment by DennisHeimbigner** [comment]
run test_raw.sh with the -x bash flag set so we can see the exact command that
is failing. Then you can use gdb on that command to see where it is crashing.

**Comment by edhartnett** [comment]
I get this:

```
(gdb) r '[dap4]file:///home/ed/unidata/netcdf-c/build/dap4_test/daptestfiles/test_anon_dim.2.syn'
Starting program: /home/ed/unidata/netcdf-c/build/ncdump/ncdump '[dap4]file:///home/ed/unidata/netcdf-c/build/dap4_test/daptestfiles/test_anon_dim.2.syn'
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
__strlen_avx2 () at ../sysdeps/x86_64/multiarch/strlen-avx2.S:62
62	../sysdeps/x86_64/multiarch/strlen-avx2.S: No such file or directory.
(gdb) up
#1  0x00007ffff7b209e1 in set_curl_options (state=0x5555557c4c40) at /home/ed/unidata/netcdf-c/libdap4/d4curlfunctions.c:203
203	        size_t hostlen = strlen(triple->host);
(gdb) p triple->host
$1 = 0x0
(gdb) p *triple
$2 = {host = 0x0, key = 0x555555786c20 "USE_CACHE", value = 0x555555786b60 "0"}
(gdb) 
```

**Comment by DennisHeimbigner** [comment]
ok, I am testing a fix. I thought I fixed this, but apparently did not
do a thorough search for all occurrences of triple->host.

**Comment by WardF** [comment]
@edhartnett 

>How exactly do you guys build with cmake, specifying the directories for HDF5? That is, can you send me what you are using from the command line?

Hi Ed,  typically I invoke as follows, from the root netcdf-c directory:

    $ mkdir build && cd build
    $ cmake .. -DCMAKE_PREFIX_PATH=$HOME/environments/local-serial [OTHER OPTIONS]

The path I specify via `CMAKE_PREFIX_PATH` contains (in this case) serial libhdf5, libhdf4, and other various and sundry installs.  Doing a directory listing of `local-serial` shows the standard `lib/`, `bin/`, `include/`, etc, directory structure.

**Comment by edhartnett** [comment]
OK thanks! I will try that next time.

**Comment by WardF** [comment]
I'm working on a presentation for COSMIC tomorrow, but I'm going to try to roll a bunch of outstanding PR's into an aggregation, this afternoon, so that we can get 4.6.1 out and move on to the next.

Mentioning it here because it's as good a place as any :)

**Comment by edhartnett** [comment]
Awesome.

**Comment by DennisHeimbigner** [comment]
This issue should be fixed by pr https://github.com/Unidata/netcdf-c/pull/888

**Comment by edhartnett** [comment]
Yes, I merged #888 into HPC netCDF and this issue is now fixed. Thanks!
